R/utilities-pk-analyses.R

```diff
@@ -23,15 +23,56 @@ calculatePKAnalyses <- function(results) {
 
 #' @title Saves the pK-analyses  to csv file
 #'
-#' @param pkAnalyses pK-Analyses to exporte (typically calculated using \code{calculatePKAnalyses} or imported from file)
+#' @param pkAnalyses pK-Analyses to export (typically calculated using \code{calculatePKAnalyses} or imported from file)
 #' @param filePath Full path where the pK-Analyses will be saved.
 #'
 #' @export
 exportPKAnalysesToCSV <- function(pkAnalyses, filePath) {
   validateIsOfType(pkAnalyses, SimulationPKAnalyses)
   validateIsString(filePath)
   filePath <- expandPath(filePath)
-  simulationExporter <- getNetTask("SimulationExporter")
-  rClr::clrCall(simulationExporter, "ExportPKAnalysesToCSV", pkAnalyses$ref, pkAnalyses$simulation$ref, filePath)
+  pkAnalysesTask <- getNetTask("PKAnalysesTask")
+  rClr::clrCall(pkAnalysesTask, "ExportPKAnalysesToCSV", pkAnalyses$ref, pkAnalyses$simulation$ref, filePath)
   invisible()
 }
+
+#' @title Loads the pK-analyses from csv file
+#'
+#' @param filePath Full path of the file containing the pK-Analyses to load.
+#' @param simulation Instance of the simulation for which the pk-Analyses were calculated. This is required to verify that the file
+#' matches the simulation
+#'
+#' @export
+importPKAnalysesFromCSV <- function(filePath, simulation) {
+  validateIsOfType(simulation, Simulation)
+  validateIsString(filePath)
+  filePath <- expandPath(filePath)
+  pkAnalysesTask <- getNetTask("PKAnalysesTask")
+  pkAnalyses <- rClr::clrCall(pkAnalysesTask, "ImportPKAnalysesFromCSV", filePath, simulation$ref)
+  SimulationPKAnalyses$new(pkAnalyses, simulation)
+}
+
+
+#' @title Convert the pk-Analysis to data frame
+#'
+#' @param pkAnalyses pK-Analyses to convert to data frame (typically calculated using \code{calculatePKAnalyses} or imported from file)
+#'
+#' @export
+pkAnalysesAsDataFrame <- function(pkAnalyses) {

```

Comment:
The methods is slightly different and requires the pkAnalysis object

Reply:
but under the hood, it does nothing more as exproted the pkanalysis to file and loading it using your code